### PR TITLE
Word wrap in code blocks

### LINKED ISSUE #2496: Word wrap in code blocks
### Question description

Hi, i really enjoy your app since i was Typora's user. But i don't see any option allowing to show code in wrapped mode inside code blocks? Thanks.

### PR DESCRIPTION
**Pull request by syco**
| Q                 | A
| ----------------- | ---
| Bug fix?          | no
| New feature?      | yes
| Breaking changes? | no
| Deprecations?     | no
| New tests added?  | not needed
| Fixed tickets     | Fixes #2496
| License           | MIT

### Description

[Description of the bug or feature]
image

### CODE REVIEW

**Comment by consal** [comment]
Is there any chance this could be moved forward? Thanks!

File: src/renderer/components/editorWithTabs/editor.vue [hunk 1]
```diff
@@ -751,6 +759,8 @@ export default {
 
       document.addEventListener('keyup', this.keyup)
 
+      setWrapCodeBlocks(wrapCodeBlocks);
```
Comment (Jocs) [inline src/renderer/components/editorWithTabs/editor.vue:762]: Please remove the semicolon.

**Comment by Jocs** [comment]
Very useful feature, I am just not sure that this feature will be implemented in the `render process`? It should still be implemented in `muya`?. From a long-term perspective, it is better to implement in `muya`, but `muya` is being refactored recently, which will lead to repeated development. If it is implemented in the `render process`, then when a new `muya` is transplanted, it will also involve the corresponding Changes. Of course, in the short term, this is not a problem. @syco @fxha  What's your opinion?

File: static/preference.json [hunk 2]
```diff
@@ -19,6 +19,7 @@
   "codeFontFamily": "DejaVu Sans Mono",
   "codeBlockLineNumbers": true,
   "trimUnnecessaryCodeBlockEmptyLines": true,
+  "wrapCodeBlocks": true,
```
Comment (fxha) [inline static/preference.json:22]: `wrapCodeBlocks` should be `false` by default to keep the old behavior.

**Comment by fxha** [comment]
> Very useful feature, I am just not sure that this feature will be implemented in the render process? It should still be implemented in muya?.

Muya for sure because the content is created and manipulated by Muya.

**Comment by Jocs** [comment]
@syco Please solve this conflicts and comments.

**Comment by syco** [comment]
Sorry, but I'm no longer using  marktext.
This has to go back to marktext's devs.
I can leave the pull request open for reference, but I'm out.

**Pull request closed without merging**